### ISSUE TITLE
Don't warn about shadowing an outer variable when it is passed as an accumulator

### ISSUE THREAD

**Issue by Drenmi** [bbatsov/rubocop#3620]
RuboCop reports an offense when passing an outer local variable to a reduce method (`#inject`, `#reduce`, `#each_with_object`) and naming the block argument with the same name.

This could be considered a false positive, because the variable and the block argument are referencing the same object. (Using the same name is actually helping make this clear.)

---
## Expected behavior

RuboCop reports no offenses.
## Actual behavior

```
tmp.rb:4:34: W: Shadowing outer local variable - bar.
foo.each_with_object(bar) do |e, bar|
                                 ^^^
```
## Steps to reproduce the problem

Run `rubocop --only Lint/ShadowingOuterLocalVariable` on:

``` ruby
bar = Hash.new { |h, k| h[k] = [] }

foo.each_with_object(bar) do |e, bar|
  bar[e] << :baz
end
```
## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.44.1 (using Parser 2.3.1.2, running on ruby 2.3.0 x86_64-darwin15)
```

**Comment by mikegee** [comment]
That's a warning from Ruby, not just a cop.

```
% ruby -wc test/test.rb
test/test.rb:3: warning: shadowing outer local variable - bar
```

**Comment by Drenmi** [comment]
@mikegee: Thinking of it some more, I guess a counter-argument can be made that one should just use `#each` instead of `#each_with_object`. WDYT?

**Comment by mikegee** [comment]
The particular example you gave would be better with `#each`. I suppose `#each` may also be a general solution to these offenses with `#inject`, `#reduce`, `#each_with_object`.

**Comment by Drenmi** [comment]
Closing this for now since Ruby considers this a warning, and you can work around it (in all cases I can come up with) by just using `#each`.
